File: app/src/main/java/com/aivoice/translate/base/PrefManager.java
package com.aivoice.translate.base;

import android.content.Context;
import android.content.SharedPreferences;

public class PrefManager {

    SharedPreferences pref;
    SharedPreferences.Editor editor;
    Context _context;

    // shared pref mode
    int PRIVATE_MODE = 0;

    // Shared preferences file name
    private static final String PREF_NAME = "speaktovoice";

    private static final String IS_FIRST_TIME_LAUNCH = "IsFirstTimeLaunch";
    private static final String IS_FIRST_TIME_INFO = "IsFirstTimeInfo";

    public PrefManager(Context context) {
        this._context = context;
        pref = _context.getSharedPreferences(PREF_NAME, PRIVATE_MODE);
        editor = pref.edit();
    }

    public void setFirstTimeLaunch(boolean isFirstTime) {
        editor.putBoolean(IS_FIRST_TIME_LAUNCH, isFirstTime);
        editor.commit();
    }

    public boolean isFirstTimeLaunch() {
        return pref.getBoolean(IS_FIRST_TIME_LAUNCH, true);
    }
    public void setFirstTimeDetail(boolean isFirstTime) {
        editor.putBoolean(IS_FIRST_TIME_INFO, isFirstTime);
        editor.commit();
    }

    public boolean isFirstTimeDetail() {
        return pref.getBoolean(IS_FIRST_TIME_INFO, true);
    }
    public void setBoolean(String dnd, boolean checked) {
        editor.putBoolean(dnd, checked);
        editor.commit();
    }
    public boolean getBoolean(String key) {
        return pref.getBoolean(key, false);
    }
    public boolean gettrueBoolean(String key) {
        return pref.getBoolean(key, true);
    }
}

File: app/src/main/java/com/texttranslate/voiceimage/base/PrefManager.kt
package com.texttranslate.voiceimage.base

import android.content.Context
import android.content.SharedPreferences

class PrefManager(_context: Context) {

    var pref: SharedPreferences = _context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    var editor: SharedPreferences.Editor = pref.edit()

    var isFirstTimeLaunch: Boolean
        get() = pref.getBoolean(IS_FIRST_TIME_LAUNCH, true)
        set(isFirstTime) {
            editor.putBoolean(IS_FIRST_TIME_LAUNCH, isFirstTime)
            editor.commit()
        }
    var isFirstTimeDetail: Boolean
        get() = pref.getBoolean(IS_FIRST_TIME_INFO, true)
        set(isFirstTime) {
            editor.putBoolean(IS_FIRST_TIME_INFO, isFirstTime)
            editor.commit()
        }

    fun setBoolean(dnd: String?, checked: Boolean) {
        editor.putBoolean(dnd, checked)
        editor.commit()
    }

    fun getBoolean(key: String?): Boolean {
        return pref.getBoolean(key, false)
    }

    companion object {
        private const val PREF_NAME = "TEXT-VOICE-TRANSLATOR"
        private const val IS_FIRST_TIME_LAUNCH = "IsFirstTimeLaunch"
        private const val IS_FIRST_TIME_INFO = "IsFirstTimeInfo"
    }
}
